python/sparkdl/graph/utils.py

```diff
@@ -0,0 +1,255 @@
+    assert len(name_parts) <= 2
+    if len(name_parts) < 2:
+        name += ":0"
+    return name
+
+def as_op_name(name):
+    """
+    Derive tf.Operation name from an op/tensor name
+
+    :param name: op name or tensor name
+    """
+    assert isinstance(name, six.string_types)
+    name_parts = name.split(":")
+    assert len(name_parts) <= 2
+    return name_parts[0]
+
+def op_name(graph, tfobj_or_name):
+    """
+    Get the name of a tf.Operation
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    return get_op(graph, tfobj_or_name).name
+
+def tensor_name(graph, tfobj_or_name):
+    """
+    Get the name of a tf.Tensor
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    return get_tensor(graph, tfobj_or_name).name
+
+def validated_output(graph, tfobj_or_name):
+    """
+    Validate and return the output names useable GraphFunction
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    return op_name(graph, tfobj_or_name)
+
+def validated_input(graph, tfobj_or_name):
+    """
+    Validate and return the input names useable GraphFunction
+
+    :param graph: tf.Graph, a TensorFlow Graph object
+    :param tfobj_or_name: either a tf.Tensor, tf.Operation or a name to either
+    """
+    graph = validated_graph(graph)
+    name = op_name(graph, tfobj_or_name)
+    op = graph.get_operation_by_name(name)
+    assert 'Placeholder' == op.type, \
+        ('input must be Placeholder, but get', op.type)
+    return name
+
+def strip_and_freeze_until(fetches, graph, sess=None, return_graph=False):
+    """
+    Converting all variables into constants
+
+    :param graph: tf.Graph, the graph to be frozen
+    :param fetches: list, graph elements representing the outputs of the graph
+    :param return_graph: bool, if set True, return the graph function object
+    :return: GraphDef, the GraphDef object with cleanup procedure applied
+    """
+    graph = validated_graph(graph)
+    should_close_session = False
+    if not sess:
+        sess = tf.Session(graph=graph)
+        should_close_session = True
+
+    gdef_frozen = tf.graph_util.convert_variables_to_constants(
+        sess,
+        graph.as_graph_def(add_shapes=True),
+        [op_name(graph, tnsr) for tnsr in fetches])
+
+    if should_close_session:
+        sess.close()
+
+    if return_graph:
+        g = tf.Graph()
+        with g.as_default():
+            tf.import_graph_def(gdef_frozen, name='')
+        return g
+    else:
+        return gdef_frozen
+
+def write_visualization_html(graph, html_file_path=None, max_const_size=32, show_in_browser=True):
```

Comment:
this is very cool! I wonder if we should advertise it more. Certainly worth of its own PR.